test/fakeserver/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+
+	uuid "github.com/nu7hatch/gouuid"
+	mvmv1 "github.com/weaveworks/flintlock/api/services/microvm/v1alpha1"
+	"github.com/weaveworks/flintlock/api/types"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func main() {
+	l, err := net.Listen("tcp", "localhost:9090")
+	if err != nil {
+		fmt.Println("Failed to listen on localhost:9090", err)
+		os.Exit(1)
+	}
+
+	s := &fakeServer{}
+
+	grpcServer := grpc.NewServer()
+	mvmv1.RegisterMicroVMServer(grpcServer, s)
+
+	if err := grpcServer.Serve(l); err != nil {
+		fmt.Println("Failed to start gRPC server", err)
+		os.Exit(1)
+	}
+}
+
+type fakeServer struct {
+	savedSpecs []*types.MicroVMSpec
+}
+
+func (s *fakeServer) CreateMicroVM(ctx context.Context, req *mvmv1.CreateMicroVMRequest) (*mvmv1.CreateMicroVMResponse, error) {
+	spec := req.Microvm
+	uid, err := uuid.NewV4()
+	if err != nil {
+		return nil, err
+	}
+
+	u := uid.String()
+	spec.Uid = &u
```

Comment:
there is a `pointyString` func in `main.go`, we could probably move that into a `pkg/utils/string.go` helper func and then change this to be something like:
```suggestion
	spec.Uid = utils.PointyString(uid.String())
```

Reply:
applied in [1b57989](https://github.com/Callisto13/hammertime/pull/5/commits/1b57989e8500c12d9b6242dfa3ee339ac402bbc6)